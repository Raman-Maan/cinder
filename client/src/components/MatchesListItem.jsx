import React from 'react';
import './styles/MatchesListItem.css';

const dateFormat = date => new Date(date).toJSON().split('T')[0];

const MatchesListItem = ({ title, subtitle, date, img }) => (
  <div className="MatchesListItem">
    <div className="img-container">
      <img className="circular" src={img} alt="" />
    </div>
    <div className="details">
      <div className="top">
        <div className="left">
          <span className="title">{title}</span>
        </div>
        <div className="right">
          <span className="date">{date ? dateFormat(date) : date}</span>
        </div>
      </div>
      <div className="bottom">
        <span className="subtitle">{subtitle}</span>
      </div>
    </div>
  </div>
);

export default MatchesListItem;
