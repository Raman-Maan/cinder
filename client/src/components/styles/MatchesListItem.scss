$item-height: 64px;
$item-padding: 7px;

$subtitle-color: #aaa;

$bg-color: #fff;
$active-bg-color: darken($bg-color, 5%);

$border-width: 65%;
$active-border-width: 90%;

$border-color: #ddd;
$active-border-color: darken($border-color, 5%);

.MatchesListItem {
  display: flex;
  position: relative;
  width: 100%;
  height: $item-height;
  background-color: $bg-color;

  transition: background-color 300ms;

  // render the underline
  &::after {
    content: '';
    position: absolute;
    width: $border-width;
    height: 1px;
    bottom: 0;
    // calculate middle value
    left: 50% - $border-width/2;
    background-color: $border-color;

    transition: all 300ms;
  }

  // don't render underline for last element
  &:last-child::after {
    width: 0;
    height: 0;
  }

  &.active {
    background-color: $active-bg-color;
    &::after {
      background-color: $active-border-color;
      width: $active-border-width;
      left: 50% - $active-border-width/2;
    }
  }

  .img-container {
    flex: 1;
    padding: $item-padding;
    height: $item-height;
    width: $item-height;

    img {
      height: 100%;
      width: 100%;

      &.circular {
        border-radius: 50%;
      }
    }
  }

  .details {
    width: calc(100% - #{$item-height});
    display: flex;
    flex-direction: column;
    padding: $item-padding;

    .top {
      display: flex;
      justify-content: space-between;

      .left {
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;

        .title {
          font-weight: bold;
        }
      }

      .right {
        white-space: nowrap;
        .date {
          color: $subtitle-color;
        }
      }
    }

    .bottom {
      text-overflow: ellipsis;
      overflow: hidden;
      white-space: nowrap;

      .subtitle {
        color: $subtitle-color;
      }
    }
  }
}
