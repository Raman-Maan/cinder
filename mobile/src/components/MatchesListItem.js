import React from 'react';
import { ListItem, Text, Left, Right, Body, Thumbnail } from 'native-base';

const dateFormat = date => new Date(date).toJSON().split('T')[0];

const MatchesListItem = ({
  userID, userName, userBio, primaryPic, matchDate, onPress,
}) => (
  <ListItem avatar onPress={onPress}>
    <Left>
      <Thumbnail source={{ uri: primaryPic }} style={{ resizeMode: 'contain' }} />
    </Left>
    <Body>
      <Text>{userName}</Text>
      <Text note numberOfLines={1}>{userBio}</Text>
    </Body>
    <Right>
      <Text note>{matchDate ? dateFormat(matchDate) : matchDate}</Text>
    </Right>
  </ListItem>
);

export default MatchesListItem;
