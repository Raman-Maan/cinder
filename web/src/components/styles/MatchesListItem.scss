$item-height: 64px;
$item-padding: 7px;

$subtitle-color: #aaa;

$focus-color: #a2a2a2;

$bg-color: #fff;
$active-bg-color: darken($bg-color, 5%);

.MatchesListItem {
  display: flex;
  position: relative;
  width: 100%;
  height: $item-height;
  background-color: $bg-color;

  transition: background-color 200ms;

  cursor: pointer;

  &.active {
    background-color: $active-bg-color;
  }

  &:hover {
    background-color: $active-bg-color;
  }

  &:focus {
    outline: none;
    box-shadow: inset 0 0 0 1px $focus-color;
    z-index: 100;
  }

  .img-container {
    flex: 1;
    padding: $item-padding;
    height: $item-height;
    width: $item-height;

    img {
      height: 100%;
      width: 100%;
      object-fit: scale-down;

      &.circular {
        border-radius: 50%;
      }
    }
  }

  .details {
    width: calc(100% - #{$item-height});
    display: flex;
    flex-direction: column;
    padding: $item-padding;

    .top {
      display: flex;
      justify-content: space-between;

      .left {
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;

        .title {
          font-weight: bold;
        }
      }

      .right {
        white-space: nowrap;
        .date {
          color: $subtitle-color;
        }
      }
    }

    .bottom {
      text-overflow: ellipsis;
      overflow: hidden;
      white-space: nowrap;

      .subtitle {
        color: $subtitle-color;
      }
    }
  }
}
