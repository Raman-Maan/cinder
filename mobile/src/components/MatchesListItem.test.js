import React from 'react';
import renderer from 'react-test-renderer';

import MatchesListItem from './MatchesListItem';

describe('<MatchesListItem />', () => {
  const testData = {
    title: 'Example title',
    subtitle: 'Some subtitle',
    date: new Date('01/01/2017'),
    img: 'img',
  };

  it('renders correctly', () => {
    const rendered = renderer.create(<MatchesListItem {...testData} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
