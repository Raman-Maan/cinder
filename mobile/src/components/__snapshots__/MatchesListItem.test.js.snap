// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchesListItem /> renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    avatar={true}
    onPress={undefined}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 15,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 0,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 28,
              "height": 56,
              "width": 56,
            },
            Object {
              "resizeMode": "contain",
            },
          ]
        }
        virtual={undefined}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
        virtual={undefined}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        numberOfLines={1}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
        virtual={undefined}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "alignSelf": null,
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "200",
          }
        }
        uppercase={false}
        virtual={undefined}
      />
    </View>
  </View>
</View>
`;
